src/wp-admin/includes/user.php

```diff
@@ -119,7 +119,13 @@ function edit_user( $user_id = 0 ) {
 		} elseif ( '' === $locale ) {
 			$locale = 'en_US';
 		} elseif ( ! in_array( $locale, get_available_languages(), true ) ) {
-			$locale = '';
+			if ( current_user_can( 'install_languages' ) && wp_can_install_language_pack() ) {
+				if ( ! wp_download_language_pack( $locale ) ) {
+					$locale = '';
```

Comment:
```suggestion
					$locale = $user->locale;
```

I think this needs to reset to the user's current locale, otherwise a failed translation installation will result in their locale being reset to the site default rather than their previously chosen. What do you think?

Reply:
I'm on the fence.

This just matches the current existing behavior to reset any invalid values.

Not sure if we wanna really change that.